pyangbind/lib/serialise.py

```diff
@@ -508,20 +509,25 @@ def generate_element(obj, parent_namespace=None, flt=False):
       generated_by = getattr(element, "_pybind_generated_by", None)
       if generated_by == "container":
         d[yname] = pybindIETFJSONEncoder.generate_element(element,
-                      parent_namespace=element._namespace, flt=flt)
+                      parent_namespace=element._namespace, flt=flt,
+                      with_defaults=with_defaults)
         if not len(d[yname]):
           del d[yname]
       elif generated_by == "YANGListType":
         d[yname] = [pybindIETFJSONEncoder.generate_element(i,
-                      parent_namespace=element._namespace, flt=flt)
+                      parent_namespace=element._namespace, flt=flt,
+                      with_defaults=with_defaults)
                         for i in element._members.itervalues()]
         if not len(d[yname]):
           del d[yname]
       else:
         if flt and element._changed():
           d[yname] = element
-        elif not flt:
-          d[yname] = element
+        else:
+          if not flt:
+            d[yname] = element
+          elif with_defaults and element._default == element:
```

Comment:
This will only include defaults when they are explicitly set AFAICS. It seems like the check that you want to do here is `elif with_defaults and not element._default == False` since this is then a leaf that is specified to have a default.

Reply:
That's exactly what I want to do, I only want to include values that were explicitly set.  How should I rename it then?

Reply:
I would suggest having an argument that takes an enumerated (`Enum` or `IntEnum`) which lets one specify the type of defaults addition that one wants. There are a few default reporting modes (e.g., those discussed in https://tools.ietf.org/html/rfc6243#section-2). I'd add an enumerated type that is something like `IF_SET` that does what you'd like. In the future we can extend this with other behaviours if they are required.